tests/test_time.py

```diff
@@ -29,7 +29,14 @@ def __init__(self, unique_id, model):
         self.steps = 0
         self.advances = 0
 
+    def kill_agent(self):
+        for agent in self.model.schedule.agents:
+            if agent != self:
+                self.model.schedule.remove(agent)
+
     def stage_one(self):
+        if self.model.calling_func == "test_intrastep_remove":
```

Comment:
Maybe define a boolean attribute `do_kill_other_agent` at agent init? `kill_other_agent` might be used in other tests.

Reply:
`enable_kill_other_agent` seems better. `do_kill_other_agent` could be erroneously interpreted as a function/action (although, `enable_kill_other_agent` also does).

Reply:
done!